src/CyclicContainer.h

```diff
@@ -0,0 +1,133 @@
+/*
+  Copyright (C) 2020 SUSE LLC
+  This library is free software; you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as
+  published by the Free Software Foundation; either version 2.1 of the
+  License, or (at your option) version 3.0 of the License. This library
+  is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+  License for more details. You should have received a copy of the GNU
+  Lesser General Public License along with this library; if not, write
+  to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+  Floor, Boston, MA 02110-1301 USA
+*/
+
+
+#ifndef CyclicContainer_h
+#define CyclicContainer_h
+
+#include <algorithm>
+#include <iterator>
+#include <vector>
+
+template <class T>
+class CyclicContainer
+{
+public:
+
+    using Iterator = typename std::vector<T *>::iterator;
+    using ReverseIterator = std::reverse_iterator<Iterator>;
+
+    Iterator begin() { return _elements.begin(); }
+    Iterator end() { return _elements.end(); }
+
+    CyclicContainer() : _elements(), _current( nullptr )
+    {}
+
+
+    ~CyclicContainer()
+    {
+	clear();
+    }
+
+
+    void clear()
+    {
+	_elements.clear();
+	_current = nullptr;
+    }
+
+
+    void add( T * element )
```

Comment:
NO! NO! NO!

This is a "naked pointer". Nobody knows who owns (deletes) the data pointed to.

Use smart pointers instead. If you must use a dumb pointer you must document the ownership.

Reply:
I'd vote to document who owns the object. That is indeed missing here.

Reply:
Since this class is ultimately using YMenuItems, ownership will remain with the YSelectionWidget, as usual; but that knowledge is implicit here. Somebody reading this code will probably not know that.

Reply:
Yes, you are right. I will add documentation.